__tests__/app.test.js

```diff
@@ -189,3 +191,58 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: Responds with the added comment", () => {
+    const newComment = { username: "butter_bridge", body: "Wow" };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(201)
+      .then((res) => {
+        expect(res.body.comment).toMatchObject({
+          article_id: 2,
+          author: newComment.username,
+          body: newComment.body,
+          comment_id: expect.any(Number),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+        });
+      });
+  });
+  test("404: Responds with User Not Found when the username is not found in the database", () => {
+    const newCommentnewUser = {
+      username: "PerdHapley",
+      body: "What I think about this article is undecided",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newCommentnewUser)
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("User Not Found");
+      });
+  });
+  test("404: Responds with Not Found when article_id is not found", () => {
+    const newComment = { username: "butter_bridge", body: "Wow" };
+
+    return request(app)
+      .post("/api/articles/200/comments")
+      .send(newComment)
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("Not Found");
+      });
+  });
```

Comment:
Really good test as well, but dont forget to include a test for when the article id given is invalid i.e. /api/articles/banana/comments